From c982cc094db557d526030ad45752c3723fc55714 Mon Sep 17 00:00:00 2001
From: Matthew Gerlach <mgerlach@altera.com>
Date: Tue, 5 Nov 2013 11:40:33 -0800
Subject: [PATCH 021/172] FogBugz #165941: FPGA bridge drivers need to be
 started early

The fpga bridge framework needs be started before the
individual bridge drivers, and the individual bridge
drivers need to be started before any other driver
for components on the FPGA.  The earliest the bridge
drivers successfully start is during arch_init().

Signed-off-by: Matthew Gerlach <mgerlach@altera.com>
[czou:Original patch taken from
https://github.com/altera-opensource/linux-socfpga.git socfpga-4.1]
Signed-off-by: czou <cao.zou@windriver.com>
---
 drivers/misc/fpga-bridge/altera-fpga2sdram.c | 2 +-
 drivers/misc/fpga-bridge/altera-hps2fpga.c   | 2 +-
 drivers/misc/fpga-bridge/fpga-bridge.c       | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/misc/fpga-bridge/altera-fpga2sdram.c b/drivers/misc/fpga-bridge/altera-fpga2sdram.c
index 213dcf7..11a7184 100644
--- a/drivers/misc/fpga-bridge/altera-fpga2sdram.c
+++ b/drivers/misc/fpga-bridge/altera-fpga2sdram.c
@@ -201,7 +201,7 @@ static void __exit alt_fpga_bridge_exit(void)
 	platform_driver_unregister(&altera_fpga_driver);
 }
 
-module_init(alt_fpga_bridge_init);
+arch_initcall(alt_fpga_bridge_init);
 module_exit(alt_fpga_bridge_exit);
 
 MODULE_DESCRIPTION("Altera SoCFPGA FPGA to SDRAM Bridge");
diff --git a/drivers/misc/fpga-bridge/altera-hps2fpga.c b/drivers/misc/fpga-bridge/altera-hps2fpga.c
index 4e5dd38..5fce8a1 100644
--- a/drivers/misc/fpga-bridge/altera-hps2fpga.c
+++ b/drivers/misc/fpga-bridge/altera-hps2fpga.c
@@ -191,7 +191,7 @@ static void __exit alt_fpga_bridge_exit(void)
 	platform_driver_unregister(&altera_fpga_driver);
 }
 
-module_init(alt_fpga_bridge_init);
+arch_initcall(alt_fpga_bridge_init);
 module_exit(alt_fpga_bridge_exit);
 
 MODULE_DESCRIPTION("Altera SoCFPGA HPS to FPGA Bridge");
diff --git a/drivers/misc/fpga-bridge/fpga-bridge.c b/drivers/misc/fpga-bridge/fpga-bridge.c
index 96bf4e4..6dcbf01 100644
--- a/drivers/misc/fpga-bridge/fpga-bridge.c
+++ b/drivers/misc/fpga-bridge/fpga-bridge.c
@@ -225,5 +225,5 @@ MODULE_DESCRIPTION("FPGA Bridge Driver");
 MODULE_AUTHOR("Alan Tull <atull@altera.com>");
 MODULE_LICENSE("GPL v2");
 
-subsys_initcall(fpga_bridge_dev_init);
+core_initcall_sync(fpga_bridge_dev_init);
 module_exit(fpga_bridge_dev_exit);
-- 
1.9.1

